docs/Features/XMLReferencesFeatures.md

```diff
@@ -0,0 +1,93 @@
+# XML References Features
+
+## References
+
+Given this [docbook](https://docbook.org/) XML file sample:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook XML V4.4//EN" "http://www.docbook.org/xml/4.4/docbookx.dtd">
+<book>
+    <chapter id="chapter-1">
+
+        <xref linkend="chapter-1" />
+
+    </chapter>
+
+    <chapter id="chapter-2">
+
+    </chapter>
+</book>
+```
+
+In this sample, `linkend` attribute in `<xref linkend="chapter-1" />` references the `chapter-1` declared in `<chapter id="chapter-1">`. [vscode-xml](https://github.com/redhat-developer/vscode-xml) provides a completion, definition, highlighting support to support this kind of references easily with the `xml.references` settings. For docbook case, you can declare this settings:
+
+```json
+  "xml.references": [
+    {
+      "pattern": "**/*.xml",
+      "expressions": [
+        {
+          "from": "xref/@linkend",
+          "to": "@id"
+        }
+      ]
+    }
+  ]
+```
+
```

Comment:
I think it would be helpful here to explain what each field means (pattern, from, to) in general terms. Ex: "pattern matches the files that this reference applies to"

Reply:
fixed